### PR TITLE
wide-ranging fixes and refactoring - includes #492 (no conflict with #491, #493)

### PR DESCRIPTION
**Pull request by fcpmanon**
**This includes an important reversal of setting $i to $slaves.length in AddSlave which overwrote $i in various multi-acquisition loops preventing acquisition of more than one slave from those events.**

1. tested starting new game without errors, buying lots of slaves, putting them into facilities or choose-own-job
2. tested weekly facility reports all run without errors when unstaffed or staffed with leaders and slaves inside
3. tested peLonelyBodyguard event: all actions worked as expected
4. tested PESS loving headgirl, worried headgirl, bodyguard bedtime, loving concubine and headgirl dickgirl: all actions worked as expected
5. tested reFullBed: all actions checked
6. tested pCoupAttempt: all actions worked as expected
7. tested fAbuse, fFeelings, fKiss, fRelation, fRival, fPCImpreg, fImpreg, fImpregConsummate: all worked
8. tested Remove activeSlave: worked as expected
9. tested specialSlave and completeCatalog: worked as expected
10. tested reRelativeRecruiter: worked as expected
11. tested peHeadgirlConcubine: worked as expected

I think this is safe to merge now.

changes (highlighted those with significant player-visible effect):

**do not change $i in AddSlave**; rename temporary variables that are used in included passages to minimize naming collisions that cause unexpected bugs in calling passages; replace various for loops with native JS array.find and array.findIndex for improved performance and reduced chance of errors (mostly focusing on tw files for events and tw files that are included by others, facility weekly reports among them); rework PartnerVCheck in miscWidgets; **artWidgets fix for prosthetics reported by anonymous blog poster**; cleanup nobr in various passages; **fix errors with shelter slaves**; various event fixes in PESS.tw; fix incorrectly setting arcology government property to an integer (should be a string) in various files; fix randomization of government types for starting arcologies; fix randomization of directions for arcologies added by cheat editing; **fix errors with traitor recovery in pCoupAttempt**; change encyclopedia link when viewing penthouse; update leader variables in asDump; **fix slaves self-assigning to master suite**; **fix corporate market being available to build in markets sectors before incorporation**; **fix reversed useTabs option checks**; **fix neighbor arcology numbering**; **fix missing _partnerSlave default assignment text in WalkPast**; improve cheatMode event selection to allow more back-and-forth navigation while testing; encode all touched files in UTF-8

### CODE REVIEW

File: src/uncategorized/pCoupAttempt.tw [hunk 1]
```diff
@@ -1,8 +1,6 @@
 :: P coup attempt [nobr]
 
-<<set $nextButton = "Continue">>
-<<set $nextLink = "Random Nonindividual Event">>
-<<set $nextButton = "Continue">>
+<<set $nextButton = "Continue, $nextLink = "Random Nonindividual Event">>
```
Comment (Folant) [inline src/uncategorized/pCoupAttempt.tw]: Missing "

File: src/uncategorized/PETS.tw [hunk 2]
```diff
@@ -2,57 +2,35 @@
...
+	''A random two slave event would have been selected from the following:''
 	<br>
 	<<for _i = 0; _i < $PETSevent.length; _i++>>
 		<<print "[[$PETSevent[_i]|PETS][$PETSevent = $PETSevent[" + _i + "]]]">>
 		<br>
 	<</for>>
+	<br><br>[[Go Back to Random Nonindividual Event|Random Nonindividual Event][$eventSlave = 0]]
   <<else>>
-	<<set $PETSevent = $PETSevent.random()>>
+	<<set $PETSevent = $PETSevent.random())>>
```
Comment (Folant) [inline src/uncategorized/PETS.tw]: Are that `)` really needed?

**Comment by fcpmanon** [comment]
@Folant Thanks for catching those mistakes.

**Comment by fcpmanon** [comment]
@Folant broken how?
image

**Comment by Folant** [comment]
@fcpmanon Well, it's not opening for me in accordion/non-acc. versions, lol. I'll try with a new save now.

**Comment by fcpmanon** [comment]
Sounds like a backwards compatibility problem.  Can you upload your save?

**Comment by Folant** [comment]
@fcpmanon Yep, here. It's working with a new save though
[free-cities-20170523-005715.zip](https://github.com/Free-Cities/Free-Cities/files/1020314/free-cities-20170523-005715.zip)

**Comment by fcpmanon** [comment]
There's a more fundamental problem with the implementation of random events -- there isn't a standard for passing parameters to the events other than global variables.

This causes problems when the same global variables are used for multiple events.  For example, like $mother/$daughter were used for both RE resistant mother daughter and RE devoted mother daughter, but if you qualified for both in the same week, you could get one event but with the globals set for the other.

I just fixed that by adding $devMother/$devDaughter, but I view that as a stopgap solution.

**Comment by fcpmanon** [comment]
@Folant Do you have a save from before you hit next turn?  The cellblockSlaves was > 0, so it showed the report, but when processing the report it found no valid slaves in $CellBiIDs and so had nothing to display.

**Comment by Folant** [comment]
No, but I guess there's another strange thing 
str

[free-cities-20170523-010907.zip](https://github.com/Free-Cities/Free-Cities/files/1020355/free-cities-20170523-010907.zip)

**Comment by fcpmanon** [comment]
@Folant That was a genuine bug.  Fixed now.  Thanks for spotting it!

**Comment by fcpmanon** [comment]
Due to the bug your cellblockSlaves counter is out of sync.  It should self-correct in the next weekly report.

**Comment by Folant** [comment]
Working fine, thanks.

**Comment by fcpmanon** [comment]
Testing my last round of PETS changes now.

**Comment by fcpmanon** [comment]
I'm done testing.  Should be OK to merge now.

**Comment by Folant** [comment]
>closing tag

🤣 
Do we have 2 superfluous closing tags somewhere?
````
$ git grep "<<\/replace" | wc -l
2451

$ git grep "<<replace" | wc -l
2449
````

**Pull request merged**